
<!DOCTYPE html>
<html lang="en">
  <head>
    <title>GTO - The Genomic Toolkit</title>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1"> 
    <!-- Latest compiled and minified CSS -->
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.1.3/css/bootstrap.min.css">

    <!-- jQuery library -->
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>

    <!-- Popper JS -->
    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.3/umd/popper.min.js"></script>

    <!-- Latest compiled JavaScript -->
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.1.3/js/bootstrap.min.js"></script> 
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
    <script>
    $(document).ready(function(){
       $('#navbar').load("/gto/navbar.html");
       $('#footer').load("/gto/footer.html");
    });
    </script>
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
    <link rel="stylesheet" href="/gto/style.css">
    <link rel="stylesheet" href="/gto/icons/styles.css">
  </head>
  <body>

    <div id="navbar"></div>
<div class="container-fluid"><br/><br/><br/>
	<h1>Program gto_genomic_compressor</h1>
<p>The <strong>gto_genomic_compressor</strong> is able to provide additional compression gains over several top specific tools, while as an analysis tool, it is able to determine absolute measures, namely for many distance computations, and local measures, such as the information content contained in each element, providing a way to quantify and locate specific genomic events.</p>
<p>For help type:</p>

			<div class="card">
				<div class="card-body">
					<code>
			  			<p>./gto_genomic_compressor -h<br></p></code></div></div><br><p>In the following subsections, we explain the input and output paramters.</p>
<p></p>
<h3>Input parameters</h3>
<p></p>
<p>The <strong>gto_genomic_compressor</strong> program needs a sequence to compress.</p>
<p>The attribution is given according to:</p>

			<div class="card">
				<div class="card-body">
					<code>
			  			<p>SYNOPSIS<br>./gto_genomic_compressor [OPTION]... -r [FILE] [FILE]:[FILE]:[FILE]:[...]<br><br>SAMPLE<br>Run Compression         :  ./gto_genomic_compressor -v -l 3 sequence.txt<br>Run Decompression       :  ./gto_genomic_decompressor -v sequence.txt.co<br>Run Information Profile :  ./gto_genomic_compressor -v -l 3 -e sequence.txt<br><br>DESCRIPTION<br>Compress and decompress genomic sequences for storage purposes.<br>Measure an upper bound of the sequences entropy.<br>Compute information profiles of genomic sequences.<br><br>-h,  --help<br>usage guide (help menu).<br><br>-V,  --version<br>Display program and version information.<br><br>-F,  --force<br>force mode. Overwrites old files.<br><br>-v,  --verbose<br>verbose mode (more information).<br><br>-x,  --examples<br>show several running examples (parameter examples).<br><br>-s,  --show-levels<br>show pre-computed compression levels (configured parameters).<br><br>-e,  --estimate<br>it creates a file with the extension ".iae" with the<br>respective information content. If the file is FASTA or<br>FASTQ it will only use the "ACGT" (genomic) sequence.<br><br>-l [NUMBER],  --level [NUMBER]<br>Compression level (integer).<br>Default level: 5.<br>It defines compressibility in balance with computational<br>resources (RAM & time). Use -s for levels perception.<br><br>-tm [NB_C]:[NB_D]:[NB_I]:[NB_H]:[NB_G]/[NB_S]:[NB_E]:[NB_A]<br>Template of a target context model.<br>Parameters:<br>[NB_C]: (integer [1;20]) order size of the regular context<br>model. Higher values use more RAM but, usually, are<br>related to a better compression score.<br>[NB_D]: (integer [1;5000]) denominator to build alpha, which<br>is a parameter estimator. Alpha is given by 1/[NB_D].<br>Higher values are usually used with higher [NB_C],<br>and related to confiant bets. When [NB_D] is one,<br>the probabilities assume a Laplacian distribution.<br>[NB_I]: (integer {0,1,2}) number to define if a sub-program<br>which addresses the specific properties of DNA<br>sequences (Inverted repeats) is used or not. The<br>number 2 turns ON this sub-program without the<br>regular context model (only inverted repeats). The<br>number 1 turns ON the sub-program using at the same<br>time the regular context model. The number 0 does<br>not contemple its use (Inverted repeats OFF). The<br>use of this sub-program increases the necessary time<br>to compress but it does not affect the RAM.<br>[NB_H]: (integer [1;254]) size of the cache-hash for deeper<br>context models, namely for [NB_C] > 14. When the<br>[NB_C] <= 14 use, for example, 1 as a default. The<br>RAM is highly dependent of this value (higher value<br>stand for higher RAM).<br>[NB_G]: (real [0;1)) real number to define gamma. This value<br>represents the decayment forgetting factor of the<br>regular context model in definition.<br>[NB_S]: (integer [0;20]) maximum number of editions allowed<br>to use a substitutional tolerant model with the same<br>memory model of the regular context model with<br>order size equal to [NB_C]. The value 0 stands for<br>turning the tolerant context model off. When the<br>model is on, it pauses when the number of editions<br>is higher that [NB_C], while it is turned on when<br>a complete match of size [NB_C] is seen again. This<br>is probabilistic-algorithmic model very usefull to<br>handle the high substitutional nature of genomic<br>sequences. When [NB_S] > 0, the compressor used more<br>processing time, but uses the same RAM and, usually,<br>achieves a substantial higher compression ratio. The<br>impact of this model is usually only noticed for<br>[NB_C] >= 14.<br>[NB_E]: (integer [1;5000]) denominator to build alpha for<br>substitutional tolerant context model. It is<br>analogous to [NB_D], however to be only used in the<br>probabilistic model for computing the statistics of<br>the substitutional tolerant context model.<br>[NB_A]: (real [0;1)) real number to define gamma. This value<br>represents the decayment forgetting factor of the<br>substitutional tolerant context model in definition.<br>Its definition and use is analogus to [NB_G].<br><br>... (you may use several target models with custom parameters)<br><br>-rm [NB_C]:[NB_D]:[NB_I]:[NB_H]:[NB_G]/[NB_S]:[NB_E]:[NB_A]<br>Template of a reference context model.<br>Use only when -r [FILE] is set (referential compression).<br>Parameters: the same as in -tm.<br><br>... (you may use several reference models with custom parameters)<br><br>-r [FILE], --reference [FILE]<br>Reference sequence filename ("-rm" are trainned here).<br>Example: -r file1.txt.<br><br>[FILE]<br>Input sequence filename (to compress) -- MANDATORY.<br>File(s) to compress (last argument).<br>For more files use splitting ":" characters.<br>Example: file1.txt:file2.txt:file3.txt.<br></p></code></div></div><br><p>In the following example, it will be downloaded seventeen DNA sequences, and compress and decompress one of the smallest (BuEb). Finally, it compares if the uncompressed sequence is equal to the original.</p>

			<div class="card">
				<div class="card-body">
					<code>
			  			<p>wget http://sweet.ua.pt/pratas/datasets/DNACorpus.zip<br>unzip DNACorpus.zip<br>cp DNACorpus/BuEb .<br>../../bin/gto_genomic_compressor -v -l 2 BuEb<br>../../bin/gto_genomic_decompressor -v BuEb.co<br>cmp BuEb BuEb.de -l<br></p></code></div></div><br></div><br/><br/><br/><br/>
    <div id="footer"></div>
  </body>
</html>
