
<!DOCTYPE html>
<html lang="en">
  <head>
    <title>GTO - The Genomic Toolkit</title>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1"> 
    <!-- Latest compiled and minified CSS -->
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.1.3/css/bootstrap.min.css">

    <!-- jQuery library -->
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>

    <!-- Popper JS -->
    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.3/umd/popper.min.js"></script>

    <!-- Latest compiled JavaScript -->
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.1.3/js/bootstrap.min.js"></script> 
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
    <script>
    $(document).ready(function(){
       $('#navbar').load("/gto/navbar.html");
       $('#footer').load("/gto/footer.html");
    });
    </script>
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
    <link rel="stylesheet" href="/gto/style.css">
    <link rel="stylesheet" href="/gto/icons/styles.css">
  </head>
  <body>

    <div id="navbar"></div>
<div class="container-fluid"><br/><br/><br/>
	<h1>Program gto_fastq_xs</h1>
<p>The <strong>gto_fastq_xs</strong> is a skilled FASTQ read simulation tool, flexible, portable (does not need a reference sequence) and tunable in terms of sequence complexity. XS handles Ion Torrent, Roche-454, Illumina and ABI-SOLiD simulation sequencing types. It has several running modes, depending on the time and memory available, and is aimed at testing computing infrastructures, namely cloud computing of large-scale projects, and testing FASTQ compression algorithms. Moreover, XS offers the possibility of simulating the three main FASTQ components individually (headers, DNA sequences and quality-scores). Quality-scores can be simulated using uniform and Gaussian distributions.</p>
<p>For help type:</p>

			<div class="card">
				<div class="card-body">
					<code>
			  			<p>./gto_fastq_xs -h<br></p></code></div></div><br><p>In the following subsections, we explain the input and output paramters.</p>
<p></p>
<h3>Input parameters</h3>
<p></p>
<p>The <strong>gto_fastq_xs</strong> program needs a FASTQ file to compute.</p>
<p>The attribution is given according to:</p>

			<div class="card">
				<div class="card-body">
					<code>
			  			<p>Usage: ./gto_fastq_xs   [OPTION]... [FILE]<br><br>System options:<br>-h                       give this help<br>-v                       verbose mode<br><br>Main FASTQ options:<br>-t  < sequencingType >     type: 1=Roche-454, 2=Illumina, 3=ABI SOLiD, 4=Ion Torrent<br>-hf < headerFormat >       header format: 1=Length appendix, 2=Pair End<br>-i  n=< instrumentName >   the unique instrument name (use n= before name)<br>-o                       use the same header in third line of the read<br>-ls < lineSize >           static line (bases/quality scores) size<br>-ld < minSize >:< maxSize >  dynamic line (bases/quality scores) size<br>-n  < numberOfReads >      number of reads per file<br><br>DNA options:<br>-f  < A >,< C >,< G >,< T >,< N >  symbols frequency<br>-rn < numberOfRepeats >    repeats: number (default: 0)<br>-ri < repeatsMinSize >     repeats: minimum size<br>-ra < repeatsMaxSize >     repeats: maximum size<br>-rm < mutationRate >       repeats: mutation frequency<br>-rr                      repeats: use reverse complement repeats<br><br>Quality scores options:<br>-qt < assignmentType >     quality scores distribution: 1=uniform, 2=gaussian<br>-qf < statsFile >          load file: mean, standard deviation (when: -qt 2)<br>-qc < template >           custom template ascii alphabet<br><br>Filtering options:<br>-eh                      excludes the use of headers from output<br>-eo                      excludes the use of optional headers (+) from output<br>-ed                      excludes the use of DNA bases from output<br>-edb                     excludes '\n' when DNA bases line size is reached<br>-es                      excludes the use of quality scores from output<br><br>Stochastic options:<br>-s  < seed >               generation seed<br><br>< genFile >                 simulated output file<br><br>Common usage:<br>./XS -v -t 1 -i n=MySeq -ld 30:80 -n 20000 -qt=1 -qc 33,36,39:43 File<br>./XS -v -ls 100 -n 10000 -eh -eo -es -edb -f 0.3,0.2,0.2,0.3,0.0 -rn 50 -ri 300 -ra 3000 -rm 0.1 File<br></p></code></div></div><br><p>An example of such an input file is:</p>

			<div class="card">
				<div class="card-body">
					<code>
			  			<p>@SRR001666.1 071112_SLXA-EAS1_s_7:5:1:817:345 length=60<br>GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG<br>+<br>IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII<br>@SRR001666.2 071112_SLXA-EAS1_s_7:5:1:801:338 length=72<br>GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCGTCGTTTTATCAT<br>+<br>IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGII>IIIII-I)8I<br></p></code></div></div><br><p></p>
<h3>Output</h3>
<p>The output of the <strong>gto_fastq_xs</strong> program is a FASTQ file</p>
<p>Using the input above using the common usage with 5 reads (-n 5), an output example for this is the following:</p>

			<div class="card">
				<div class="card-body">
					<code>
			  			<p>@output.fastq.598 LQGQLWH01D5WVZ length=62<br>TTCNTNCCAGGTAAAGAGAACATNCCGNCGCACTACTCGTAAGACTTGCTGGNCGAGAAAGG<br>+<br>)(+!*!$')($(()+'))$$()'!)!$!!$*+)+''('!)))!+!)(!+!*$!'$*)**++!<br>@output.fastq.1510 LQGQLWH01A7LJI length=57<br>CTAGACTACTCGAGCACTAGGCTCGCGTNTACCANGGGGNCTGCGNGTTGGCNCGGT<br>+<br>)+(*(+$*)+!*)!'!!(!(!!(*'$!+!(()$'!!+*+!!))!*!')***+!$+''<br>@output.fastq.2153 LQGQLWH01CHBQJ length=33<br>ACTTTTTGCTCAAGCAGGGTTGCCTAGCAANAC<br>+<br>*)++!+$''')*)**!+)$(*((*)$!'!+!!*<br>@output.fastq.3251 LQGQLWH01C8OY4 length=75<br>TCTTTCCTTCNCGNCCNAATTCCCCATAANAACTTAAAATCNCNNGCTGCGCGTGATCAACAATATTAATACTCC<br>+<br>!*''+*'!''!+!!!*!'!+(++)*(*($!!*((')$*!$(!'!!'+)$+*!$*!**!'()$!*'+'*'+!!+'(<br>@output.fastq.3934 LQGQLWH01AQDXM length=36<br>GGTAACNNGGAATTCTTCCAATTANCCNTGTCCGGC<br>+<br>$+)'!'!!)+)+!''**$$*!!')!+)!)*()!))$<br></p></code></div></div><br></div><br/><br/><br/><br/>
    <div id="footer"></div>
  </body>
</html>
